src/lazy/abstract.ts

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+export type SelectArray<T, TLazy extends 1 | 0> = TLazy extends 1 ? ILazyArray<T> : ILazyArray<T>;
+
+export type MapFn<T, TIndex, U> = (t: T, index: TIndex) => U;
+export type FilterFn<T, TIndex> = (t: T, index: TIndex) => boolean;
+export type FlatMapFnRT<T, TLazy extends 1 | 0> = T | T[] | ILazySet<T> | SelectArray<T, TLazy> | Set<T>;
+export type FlatMapFn<T, TIndex, U, TLazy extends 1 | 0 = 0> = (t: T, index: TIndex) => FlatMapFnRT<U, TLazy>;
+export type EachFn<T, TIndex> = (t: T, index: TIndex) => unknown | void | Promise<unknown> | Promise<void>;
+
+export type ObjectKey = string | symbol | number;
+
+export type ObjectKeyedIterable<T, TKey extends ObjectKey> = ILazyObject<T, TKey> | Map<TKey, T> | Record<TKey, T>;
+export type SimpleKeyedIterable<T> = ILazySet<T> | ILazyArray<T> | T[] | Set<T> | Iterable<T>;
+export type SeedIterable<T, TKey extends ObjectKey> = SimpleKeyedIterable<T> | ObjectKeyedIterable<T, TKey>;
+
+export type IKeyedLazy<T, TKey extends ObjectKey> = ILazyObject<T, TKey> | ILazyArray<T>;
+export type IAnyLazy<T, TKey extends ObjectKey = never> = IKeyedLazy<T, TKey> | ILazySet<T> | ILazyValue<T>;
+
+export type SelectCollected<T, TType extends Type, TKey extends ObjectKey = never> = TType extends Type.Set
+	? Set<T>
+	: TType extends Type.Array
+	? // ? ICollectedArray<T>
+	  T[]
+	: TType extends Type.Object
+	? Record<TKey, T>
+	: TType extends Type.Value
+	? T
+	: never;
```

Comment:
export type SelectCollected<
  T,
  TType extends Type,
  TKey extends ObjectKey = never
> = TType extends Type.Set
  ? Set<T>
  : TType extends Type.Array
  ? T[]
  : TType extends Type.Object
  ? Record<TKey, T>
  : TType extends Type.Value
  ? T
  : never;
```suggestion
export type SelectCollected<T, TType extends Type, TKey extends ObjectKey = never> = TType extends Type.Set
	? Set<T>
	: TType extends Type.Array
	? // ? ICollectedArray<T>
	  T[]
	: TType extends Type.Object
	? Record<TKey, T>
	: TType extends Type.Value
	? T
	: never;

```

Little bit easier to read imo, feel free to ignore

Reply:
lmao GitHub formatted it exactly the same

Reply:
I don't think I understand what the difference is 🙃

Reply:
There is no difference apparently lol 
there was a difference in my editor 🤔 let's forget this